specifyweb/frontend/js_src/lib/hooks/useResourceValue.tsx

```diff
@@ -240,5 +241,6 @@ export function useResourceValue<
     validationRef,
     setValidation,
```

Comment:
hmm, what I had in mind would be more like this:

```suggestion
    setValidation: React.useCallback((messages)=>setBlockers(messages, 'validation'), [setBlockers]),
```

this avoids the need to add new props (and the confusion of `do I pick setBlockers or setValidation?) and brings this bug fix to all existing code too

Reply:
Using 'validation' as the key every time doesn't work well as I need to set blockers on longitude as well as latitude. If I use the same key, it will remove any existing blocker from the other field. Also, `setValidation`'s signature comes from the return type of `useValidation`:
https://github.com/specify/specify7/blob/43c4b7484498e32521e123591794ea06b4614dc1/specifyweb/frontend/js_src/lib/hooks/useResourceValue.tsx#L56-L64

and so the `message` param could either be an array or a string. Removing `ReturnType<typeof useValidation>` and changing the signature for `useValidation` breaks typecheck in a lot of other places. The `field` prop for `useResourceValue` accepts undefined and so I've added a check for that in `setValidation` and so this is what I came up with in the end:

https://github.com/specify/specify7/blob/de082ea89cceb1a82a041e691df651fbffbe0c76/specifyweb/frontend/js_src/lib/hooks/useResourceValue.tsx#L241-L247